mantidimaging/gui/windows/nexus_load_dialog/presenter.py

```diff
@@ -274,12 +279,15 @@ def _get_images(self, image_key_number: ImageKeys, before: bool | None = None) -
         assert self.image_key_dataset is not None and self.image_key_dataset.size is not None
         assert self.data is not None
         if image_key_number is ImageKeys.Projections:
-            indices = self.image_key_dataset[...] == image_key_number.value
+            indices = np.array(self.image_key_dataset) == image_key_number.value
         else:
             if before:
-                indices = self.image_key_dataset[:self.image_key_dataset.size // 2] == image_key_number.value
+                image_key_array = np.array(self.image_key_dataset)
+                indices = image_key_array[:image_key_array.size // 2] == image_key_number.value
             else:
+                # indices = np.array(self.image_key_dataset) == image_key_number.value
                 indices = self.image_key_dataset[:] == image_key_number.value
+
```

Comment:
Might not need these if the type of `self.image_key_dataset` is specified in the constructor.